pkg/providers/instance/instance.go

```diff
@@ -450,6 +498,34 @@ func filterUnwantedSpot(instanceTypes []*cloudprovider.InstanceType) []*cloudpro
 	return instanceTypes
 }
 
+// filterOnlyCapacityReservations is used to filter only offerings that are capacity reservations
+func filterOnlyCapacityReservations(instanceTypes []*cloudprovider.InstanceType) []*cloudprovider.InstanceType {
```

Comment:
I think there may be some helper functions in the `cloudprovider` package `types.go` file that might be able to help you here